addons/account/tests/test_account_inalterable_hash.py

```diff
@@ -0,0 +1,182 @@
+        moves = (
+            self.init_invoice("out_invoice", self.partner_a, "2023-01-01", amounts=[1000, 2000])
+            | self.init_invoice("out_invoice", self.partner_b, "2023-01-02", amounts=[1000, 2000])
+        )
+        moves.action_post()
+
+        # No records to be hashed because the restrict mode is not activated yet
+        integrity_check = moves.company_id._check_hash_integrity()['results'][0]  # First journal
+        self.assertEqual(integrity_check['msg_cover'], 'This journal is not in strict mode.')
+
+        # No records to be hashed even if the restrict mode is activated because the hashing is not retroactive
+        self.company_data['default_journal_sale'].restrict_mode_hash_table = True
+        integrity_check = moves.company_id._check_hash_integrity()['results'][0]
+        self.assertEqual(integrity_check['msg_cover'], 'There isn\'t any journal entry flagged for data inalterability yet for this journal.')
+
+        # Everything should be correctly hashed and verified
+        new_moves = (
+            self.init_invoice("out_invoice", self.partner_a, "2023-01-03", amounts=[1000, 2000])
+            | self.init_invoice("out_invoice", self.partner_b, "2023-01-04", amounts=[1000, 2000])
+            | self.init_invoice("out_invoice", self.partner_a, "2023-01-05", amounts=[1000, 2000])
+            | self.init_invoice("out_invoice", self.partner_b, "2023-01-06", amounts=[1000, 2000])
+            | self.init_invoice("out_invoice", self.partner_a, "2023-01-07", amounts=[1000, 2000])
+        )
+        new_moves.action_post()
+        moves |= new_moves
+        integrity_check = moves.company_id._check_hash_integrity()['results'][0]
+        self.assertRegex(integrity_check['msg_cover'], f'Entries are hashed from {moves[2].name}.*')
+        self.assertEqual(integrity_check['first_move_date'], format_date(self.env, fields.Date.to_string(moves[2].date)))
+        self.assertEqual(integrity_check['last_move_date'], format_date(self.env, fields.Date.to_string(moves[-1].date)))
+
+        # Let's change one of the fields used by the hash. It should be detected by the integrity report.
+        # We need to bypass the write method of account.move to do so.
+        Model.write(moves[4], {'date': fields.Date.from_string('2023-01-07')})
+        integrity_check = moves.company_id._check_hash_integrity()['results'][0]
+        self.assertEqual(integrity_check['msg_cover'], f'Corrupted data on journal entry with id {moves[4].id}.')
+
+        # Let's try with the inalterable_hash field itself
+        Model.write(moves[4], {'date': fields.Date.from_string("2023-01-05")})  # Revert the previous change
+        Model.write(moves[-1], {'inalterable_hash': 'fake_hash'})
+        integrity_check = moves.company_id._check_hash_integrity()['results'][0]
+        self.assertEqual(integrity_check['msg_cover'], f'Corrupted data on journal entry with id {moves[-1].id}.')
+
+    def test_account_move_hash_versioning_1(self):
```

Comment:
I don't get what this is doing other than `test_account_move_inalterable_hash` and `test_account_move_hash_integrity_report` or even `test_account_move_hash_versioning_mix`
Same for `test_account_move_hash_versioning_2`